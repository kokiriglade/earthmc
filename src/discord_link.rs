//! # Discord Link
//!
//! Defines the [`DiscordLink`] struct.
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Serialize, Deserialize, Debug, PartialEq)]
pub struct DiscordLink {
    /// Discord User ID.
    pub id: String,
    /// Minecraft UUID.
    pub uuid: Uuid,
}
