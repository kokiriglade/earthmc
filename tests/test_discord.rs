use earthmc::discord_link::DiscordLink;

#[test]
fn test_location() {
    let raw_json = include_str!("inputs/discord.json");
    let parsed: Vec<DiscordLink> = serde_json::from_str(raw_json).unwrap();
    insta::assert_debug_snapshot!(parsed);
}
