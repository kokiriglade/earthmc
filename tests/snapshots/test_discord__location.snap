---
source: tests/test_discord.rs
expression: parsed
---
[
    DiscordLink {
        id: "160374716928884736",
        uuid: fed0ec4a-f1ad-4b97-9443-876391668b34,
    },
    DiscordLink {
        id: "160374716928884736",
        uuid: f17d77ab-aed4-44e7-96ef-ec9cd473eda3,
    },
]
